metricbeat/mb/event.go

```diff
@@ -27,7 +27,7 @@ type Event struct {
 
 // BeatEvent returns a new beat.Event containing the data this Event. It does
 // mutate the underlying data in the Event.
-func (e *Event) BeatEvent(module, metricSet string, modifiers ...EventModifier) beat.Event {
+func (e *Event) BeatEvent(module, metricSet string, period time.Duration, modifiers ...EventModifier) beat.Event {
```

Comment:
Instead of a new parameter here, I think a `Period` field inside of `Event` might be better. If `Period` is not set then the configured period is used.